core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -100,6 +102,30 @@ public static String propertyAsString(
     return defaultValue;
   }
 
+  /**
+   * Validate the table commit related properties to have non-negative integer on table creation to
+   * prevent commit failure
+   *
+   * @param properties input map
+   */
+  public static void validateCommitProperties(Map<String, String> properties) {
+    for (String commitProperty : TableProperties.COMMIT_PROPERTIES) {
```

Comment:
entirely stylistic, will stream, filter be cleaner?

Reply:
Actually with exception handling and validation throwing new exceptions in the code, I think for each loop shall be fine. Let me know if you hate it